jwt/algorithms.py

```diff
@@ -142,7 +142,7 @@ class NoneAlgorithm(Algorithm):
     """
 
     def prepare_key(self, key):
-        if key == "":
+        if key in ("", b""):
```

Comment:
When Python is executed with `-b`, this will emit a `BytesWarning` as it is comparing str to bytes.

Reply:
Does this `BytesWarning` have a deeper meaning or was is created with the transition from Python 2 to 3 in mind?

Should we generalize it to the following or does that have any relevant downsides?
```suggestion
        if not key:
```

Or this?
```suggestion
        if isinstance(key, (str, bytes)) and not key:
```

Reply:
@michael-k I think `if not key` is better choice because it's simple and can cover wider cases.